protostar/commands/test/starkware/execution_resources_summary.py

```diff
@@ -1,19 +1,114 @@
-from dataclasses import dataclass
-from typing import Dict
+import dataclasses
+import statistics
+from abc import ABC, abstractmethod
+from collections import defaultdict
+from dataclasses import dataclass, field
+from typing import Dict, List, Iterable, Optional
+from typing_extensions import Self
 
 from starkware.cairo.lang.vm.cairo_pie import ExecutionResources
 
 
+class Statistic(ABC):
+    @abstractmethod
+    def __str__(self) -> str:
+        ...
+
+    @abstractmethod
+    def __bool__(self) -> bool:
+        ...
+
+    def add_observation(self, other: "Statistic") -> "CountSeriesStatistic":
```

Comment:
I still think It would be better to explitly create empty series but right now it's fine enough

Reply:
I don't understand :|